### DISCUSSION TITLE
PHP8.4 RFC for Resource to Object Conversion

### DISCUSSION THREAD

**Discussion by valzargaming** [reactphp/reactphp#559] [Ideas]
is_resource() will soon be deprecated and resources will be treated as objects. There's a few places in the stream repository that still use it, and official recommendations have been made to address compatibility: https://wiki.php.net/rfc/resource_to_object_conversion
![image](https://github.com/reactphp/reactphp/assets/7202504/8d61e513-70d7-46f7-946d-974ddfe4d168)

**Comment by SimonFrings** [comment]
Hey @valzargaming, very good input, thanks for bringing this up :+1:

I really appreciate your ticket to raise awareness that we will encounter a deprecation in PHP 8.4. The new version of PHP is obviously still under development, and there will likely be more deprecations that could affect us. However, we still have until November to address the PHP 8.4 compatibility.

As you may have already seen in https://github.com/reactphp/event-loop/issues/271, we're moving more and more towards ReactPHP v3 (which is planned to go live around summer this year). This is currently what we're focusing on and we will probably look into PHP 8.4 compatibility in the second half of 2024.

Nonetheless, we're always happy about contributions, so if you have any ideas on how to make this work across our range of supported PHP versions (PHP 5+ for v1), I'm interested to take a look at a PR.

Hope this explains our rough roadmap for PHP 8.4 :rocket:
